README.md

```diff
@@ -1,78 +1,406 @@
+When the above command has finished, you'll find that the generator has created an `out` directory which has four files:
 
-See [additional details here](./docs/Options/GenerateOptions.md) on the full list of switches that can be provided.
+~~~
+$ ls out
+1.csv           2.csv           3.csv           manifest.json
+~~~
 
-## How it works
+The manifest file details which rules have been violated in each of the output files:
 
-There are 2 discrete primary elements:
+~~~
+{
+  "cases": [
+    {
+      "filePath": "1",
+      "violatedRules": ["first name"]
+    },
+    {
+      "filePath": "2",
+      "violatedRules": ["age"]
+    },
+    {
+      "filePath": "3",
+      "violatedRules": ["national insurance"]
+    }
+  ]
+}
+~~~
 
-| Product | Description | Status | Notes |
-| ---- | ---- | ---- | ---- |
-| Profile | A representation of the data schema and rules which can be used to generate data | Stable | |
-| Generator | A tool for generating data from a given profile | Stable - beta | Supports data generation and generation of data that has been [delibrately violated](./generator/docs/DeliberateViolation.md). |
+If you take a look at the generated file `1.csv` you'll see that data for the `firstName` field does not objy the given constraints, whilst those for `age` anf `nationalInsurance` are correct:
 
-See [overview](./docs/Profiles.md) and [schema](./docs/Schema.md) for details about profiles and how they work.
+~~~
+firstName,age,nationalInsurance
+-619248029,71,"HT849919"
+08-12-2001 02:53:16,15,"N/A"
+"Lorem Ipsum",11,"N/A"
+"Lorem Ipsum",71,"WX004081"
+1263483797,19,"HG054666"
+,75,"GE023082"
+"Lorem Ipsum",59,"ZM850737C"
+[...]
+~~~
 
-* [Generator](./generator/README.md)
-  * [Generation algorithm](./generator/docs/GenerationAlgorithm.md)
-  * [Set data generation process](./generator/docs/SetRestrictionAndGeneration.md)
-  * [Decision trees](./docs/DecisionTrees/DecisionTrees.md)
+However, it might be a surprise to see nulls, numbers and dates as values for the `firstName` field alongside strings that do not match the regex given in the profile. This is because these are all defined as a single rule within the profile. You have a couple of options if you want to ensure that `firstName` is not null and a string, the first is to inform the generator that it should not violate specific constraint types:
 
-* [Frequently asked questions](docs/FrequentlyAskedQuestions.md)
+~~~
+$ java -jar generator.jar generate --violate --dont-violate=ofType null \
+  --max-rows=100 --overwrite profile.json out
+~~~
 
-## Contributing
+Or, alternatively, you can re-arrange your constraints so that the ones that define types / null, are grouped as a single rule. After By re-grouping constraints, the following output, with random strings that violate the regex constraint, is generated:
 
-See [contributing](./.github/CONTRIBUTING.md) guidance and the [key decisions log](docs/KeyDecisions.md).
+~~~
+firstName,age,nationalInsurance
```

Comment:
TODO: encoding issues! Github failed to render this file as markdown, so I tried to replace chars with entities. However, as this is a fenced code block, these entities were not translated.

This does open up a bigger discussion. What if I want violating data, but not from such a broad range of unicode characters?

Reply:
See #294 for discussion around this topic, it needs plenty of additional thought though